frontend/src/views/instructors/editable-instructors-list.js

```diff
@@ -65,7 +118,22 @@ function EditableInstructorsList(props) {
         },
     ];
 
-    return <InstructorsList columns={columns} {...rest} />;
+    return (
+        <React.Fragment>
+            <InstructorsList columns={columns} {...rest} />
+            <DeleteInstructorDialog
+                show={deleteDialogVisible}
+                onHide={() => {
+                    setDeleteDialogVisible(false);
```

Comment:
Instead of using an effect on the dialog visibility, just make a callback here that does everything you want.